lib/Migration/Version03009Date20231114150437.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Julius Härtl <jus@bitgrid.net>
+ *
+ * @authorJulius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Text\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version03009Date20231114150437 extends SimpleMigrationStep {
```

Comment:
Apparently the version schema is different from what occ migration:generate suggest. It should be 3009 starting with that now would cause the migrations on new setups to run in the wrong order.

@nickvergessen Do you see any issue with continuing with the 0 padded major version? Is there any possible migration path to e.g. merge existing migration steps into one matching the new schema?

Reply:
Shouldn't this version string actually start with `030901` given the version 3.9.1?

Reply:
No just major/minor:

> We recommend to use the major and minor digits of your apps version for that. This allows you to introduce a new migration in your branch for a Nextcloud version if there is already a migration path for a newer one in another branch. Since you can’t change this retroactive, we recommend to leave enough space in between and therefore map the numbers to 3 digits: `1.0.x => 1000, 2.34.x => 2034, etc.

https://docs.nextcloud.com/server/latest/developer_manual/basics/storage/migrations.html

Reply:
Ah it only gets executed first because I also overseen the extra minor step that I added. However looking at https://github.com/nextcloud/server/blob/6f39d82031202a2a8d1fb81ea448a2ffdc52cc29/lib/private/DB/MigrationService.php#L223-L231 i think it should be fine to just continue with our current pattern

Reply:
Yeah should work